devito/stencilkernel.py

```diff
@@ -217,6 +218,70 @@ def apply(self, *args, **kwargs):
 
         return summary
 
+    def argument_list(self, **kwargs):
```

Comment:
This is not duplicated, data initialization is also removed and dim_size check are different